src/backends/fixerio/index.js

```diff
@@ -114,14 +117,21 @@ class FixerIoBackend {
     //
     let rate = new BigNumber(destinationRate).div(sourceRate)
     rate = this._subtractSpread(rate)
-
-    const sourceScale = this.getInfo(params.source_ledger).currencyScale
-    const destinationScale = this.getInfo(params.destination_ledger).currencyScale
-    const sourceAmount = new BigNumber(PROBE_SOURCE_AMOUNT).shift(sourceScale)
-    const destinationAmount = new BigNumber(sourceAmount)
-      .times(rate)
-      .shift(destinationScale - sourceScale)
-    return { points: [[0, 0], [sourceAmount.toNumber(), destinationAmount.toNumber()]] }
+      .shift(destinationInfo.currencyScale - sourceInfo.currencyScale)
+
+    return (typeof sourceInfo.maxBalance === 'undefined'
+      ? Promise.resolve()
+      : sourceInfo.getBalance()
+        .then(balance => [ sourceInfo.maxBalance - balance, Math.floor((sourceInfo.maxBalance - balance) * rate) ])
```

Comment:
Given that `getCurve` is a generator, I think it would be easier to understand if you `yield`ed the promise calls and used if statements instead of ternary expressions.